rebasehelper/spec_content.py

```diff
@@ -168,7 +172,10 @@ def _split_sections(cls, content):
         sections = [('%package', lines[:section_beginnings[0]])]
 
         for i in range(len(section_beginnings) - 1):
-            start = section_beginnings[i] + 1
+            start = section_beginnings[i]
+            if start is None:
+                continue
+            start += 1
```

Comment:
This is rather out of scope of this commit. It may be a bug that was revealed by type checking, but that doesn't mean it's related to it.

Reply:
I don't think that's a bug... `None` will always be the last element and not any other. And as you can see, the `for` loop iterates only through N-1 elements, so `section_beginnings[i]` cannot be `None`. There doesn't seem to be a way how to describe this in type checking. `section_beginnings` has to be `List[Optional[int]]` since we are appending None and due to being `Optional`, we have to check for None in the loop...

We could get rid of appending `None` but then it would require an if-else in the loop (different behaviour at the end of the list).

Reply:
Ok, then I think adding `mypy: ignore` and a comment with explanation would be better.